src/carve/main.clj

```diff
@@ -43,12 +43,18 @@
   (when-not (every? valid-path? paths)
     (throw (ex-info "Path not found" {:paths paths}))))
 
-(defn -main [& [flag opts & _args]]
+(defn main
+  [& [flag opts & _args]]
   (when-not (= "--opts" flag)
     (throw (ex-info (str "Unrecognized option: " flag) {:flag flag})))
   (let [opts (edn/read-string opts)
         _ (validate-opts! opts)
+        format (-> opts :report :format)
         report (impl/run! opts)]
-    (when-let [r (:report opts)]
-      (let [format (:format r)]
-        (impl/print-report report format)))))
+    (when format
+      (impl/print-report report format))
+    (if report 1 0)))
```

Comment:
You are relying on `report` being something truthy. It it ever `nil`? Shouldn't you return `0` if the report is empy?

Reply:
I think it would also be good to add unit tests for testing `0` or `1` for a zero and a positive amount of findings.

Reply:
Yes sorry you're right, I fixed it and added a couple of tests.